go/vt/sqlparser/sql_test.go

```diff
@@ -0,0 +1,323 @@
+				Columns: []*IndexColumn{
+					{Column: ColIdent{val: "col1"}},
+				},
+				Options: []*IndexOption{},
+			},
+		},
+		{
+			"CREATE UNIQUE INDEX wxyzIndex ON asdf (w,X, y , z)",
+			TableName{
+				Name: TableIdent{"asdf"},
+				Qualifier: TableIdent{},
+			},
+			IndexAlteration{
+				Action: CreateStr,
+				FromName: ColIdent{},
+				ToName: ColIdent{val: "wxyzIndex"},
+				Using: ColIdent{},
+				Type: "unique",
+				Columns: []*IndexColumn{
+					{Column: ColIdent{val: "w"}},
+					{Column: ColIdent{val: "X"}},
+					{Column: ColIdent{val: "y"}},
+					{Column: ColIdent{val: "z"}},
+				},
+				Options: []*IndexOption{},
+			},
+		},
+		{
+			"CREATE SPATIAL INDEX bestindex ON besttable (bestcol) COMMENT 'hello world'",
+			TableName{
+				Name: TableIdent{"besttable"},
+				Qualifier: TableIdent{},
+			},
+			IndexAlteration{
+				Action: CreateStr,
+				FromName: ColIdent{},
+				ToName: ColIdent{val: "bestindex"},
+				Using: ColIdent{},
+				Type: "spatial",
+				Columns: []*IndexColumn{
+					{Column: ColIdent{val: "bestcol"}},
+				},
+				Options: []*IndexOption{{Name: "comment", Value: &SQLVal{Type:StrVal, Val: []byte("hello world")}}},
+			},
+		},
+		{
+			"CREATE INDEX sOmEiNdEx USING BTREE ON tableBOI (ye, ne) USING HASH", // doesn't make sense but valid
+			TableName{
+				Name: TableIdent{"tableBOI"},
+				Qualifier: TableIdent{},
+			},
+			IndexAlteration{
+				Action: CreateStr,
+				FromName: ColIdent{},
+				ToName: ColIdent{val: "sOmEiNdEx"},
+				Using: ColIdent{val: "btree"},
+				Type: "",
+				Columns: []*IndexColumn{
+					{Column: ColIdent{val: "ye"}},
+					{Column: ColIdent{val: "ne"}},
+				},
+				Options: []*IndexOption{{Name: "using", Using: "HASH"}},
+			},
+		},
+	}
+	testIndex(t, tests)
+}
+
+func TestDropIndex(t *testing.T) {
+	tests := []testIndexStruct{
+		{
+			"DROP INDEX idx_v ON tbl_abc",
+			TableName{
+				Name: TableIdent{"tbl_abc"},
+				Qualifier: TableIdent{},
+			},
+			IndexAlteration{
+				Action: DropStr,
+				FromName: ColIdent{},
+				ToName: ColIdent{val: "idx_v"},
+				Using: ColIdent{},
+				Type: "",
+				Columns: []*IndexColumn{},
+				Options: []*IndexOption{},
+			},
+		},
+		{
+			"DROP INDEX otherName ON tableYeah",
+			TableName{
+				Name: TableIdent{"tableYeah"},
+				Qualifier: TableIdent{},
+			},
+			IndexAlteration{
+				Action: DropStr,
+				FromName: ColIdent{},
+				ToName: ColIdent{val: "otherName"},
+				Using: ColIdent{},
+				Type: "",
+				Columns: []*IndexColumn{},
+				Options: []*IndexOption{},
+			},
+		},
+	}
+	testIndex(t, tests)
+}
+
+type testIndexStruct struct {
+	statement string
+	resTable TableName
+	res IndexAlteration
+}
+func testIndex(t *testing.T, tests []testIndexStruct) {
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%v", test.statement), func(t *testing.T) {
+			res, err := Parse(test.statement)
+			require.NoError(t, err)
+			ddlRes, ok := res.(*DDL)
+			require.True(t, ok)
+			assert.Equal(t, AlterStr, ddlRes.Action)
```

Comment:
You should be able to just use assert.Equal on the expected and actual structs, no need to compare every field explicitly like this.